### ISSUE TITLE
Issue with JSONSerialization

### ISSUE THREAD

**Issue by joeosburn** [Frizlab/BSONSerialization#3]
There appears to be an issue with how BSONSerialization interacts with Foundation's JSONSerialization (https://developer.apple.com/documentation/foundation/jsonserialization/1415493-jsonobject)

When a JSON String is parsed into a Dictionary via the method `JSONSerialization.jsonObject`, if the JSON String contains an int value, JSONSerialization parses it into an NSNumber value instead of into a Boolean value. The problem is that this can be cast as a Int, Double, or a Boolean. Since the `write(BSONEntity...)` method first attempts to cast the entity as a boolean, it will succeed, even if the value is an Int, and it gets written out as a Boolean. For example, you can run this from the repl:

```
import Foundation
import BSONSerialization

let jsonString = "{\"intProperty\": 0}"
let dict = try! JSONSerialization.jsonObject(with: jsonString.data(using: .utf8)!, options: []) as! [String: Any]
let serializedBSONDoc = try BSONSerialization.data(withBSONObject: dict, options: [])
let unserializedBSONDoc = try BSONSerialization.bsonObject(with: serializedBSONDoc, options: [])
->
unserializedBSONDoc: BSONDoc = 1 key/value pair {
  [0] = {
    key = "intProperty"
    value = false
  }
}
```

Please see this Stackoverflow for more details: 
https://stackoverflow.com/questions/49641102/using-jsonserialization-to-dynamically-figure-out-boolean-values

I will submit a PR shortly with one possible solution.

**Comment by Frizlab** [comment]
Well spotted, and thanks for the PR! I commented the PR directly.

**Comment by Frizlab** [comment]
Should be fixed by #4.
